components/02-molecules/edit/FieldsContext.tsx

```diff
@@ -304,43 +304,48 @@ const FieldsProvider: React.FC<FieldsProviderProps> = ({ children }) => {
     },
   ]);
 
-  const updateFieldsWithEnsData = (ensData: ResolvedEnsData | null) => {
-    if (!ensData) {
+  const updateEditModalFieldsWithEnsData = (domainData: DomainData | null) => {
+    if (!domainData) {
       console.warn("FieldsContext - updateFieldsWithEnsData - No ENS Data");
       return;
     }
-    if (!ensData.texts || _.isEmpty(ensData.texts)) {
+
+    const { texts, addresses } = domainData.resolver;
+    if (!texts || _.isEmpty(texts)) {
       console.warn(
         "FieldsContext - updateFieldsWithEnsData - Empty ENS Data texts"
       );
     }
-    const textsKeys = Object.keys(ensData.texts || {});
-    const coinNames = ensData.coins?.map((coin) => coin.name) ?? [];
+    const textsKeys = Object.keys(texts || {});
+    const coinNames = !addresses ? [] : addresses.map((coin) => coin.label);
+
     const newProfileFields: Field[] = profileFields.map((field) => {
       if (textsKeys.includes(field.label)) {
         return {
           ...field,
-          value: (ensData.texts as TextRecords)[field.label] as string,
+          value: String(texts[field.label]),
         };
       }
       return field;
     });
-    const newAddressesFields = addressesFields.map((field) => {
-      if (coinNames.includes(field.label)) {
+    const newAddressesFields = addressesFields.map((addressField) => {
+      if (coinNames.includes(addressField.label)) {
+        const newAddress = addresses.find(
+          (address) => address.label === addressField.label
+        )?.address as string;
```

Comment:
```suggestion
        const inputtedNewAddressForLabel = addresses.find(
          (address) => address.label === addressField.label
        );
        const addressValue = inputtedNewAddressForLabel || "" 
```

Reply:
@FrancoAguzzi Please be more clear on future request changes. I've changed the function name. 

the name `inputtedNewAddressForLabel` makes no sense in this context. The function responsibility is to populate the addresses tab context with values from the domain data, not inputted addresses.

Now I've created that fallback value to fix this 🤝🏻

Reply:
@eduramme as done in the other PR and in other previous:

1. We **do not** want type aliasing. Whenever a type is aliased it shall be nice to say why (either with an explainable doc around why you are even needing to use this typescript joker or with `|| ""`)
2. We do not want `?.` spread into the code so, the suggestion above is a conditional for wether the address is found or not, to set its value based on this.

The code shared above is pretty and well written, IMO

Yes, I might have mixed contexts on the name convention, please update it to `addressesFieldsBasedOnDomainExistingData` or something like

And sorry, what is the fallback value mentioned? Didn't catch it in code